sensor_test.py

```diff
@@ -0,0 +1,167 @@
+
+    print("Step 3", 'Validate that current sensor name matches the name set in Step 1')
+    assert sensor_info_after_name_change.get("name") == updated_sensor_name
+
+
+def test_set_sensor_reading_interval(
+        get_sensor_info, set_sensor_reading_interval, get_sensor_reading
+):
+    """
+    1. Set sensor reading interval to 1.
+    2. Get sensor info.
+    3. Validate that sensor reading interval is set to interval from Step 1.
+    4. Get sensor reading.
+    5. Wait for interval specified in Step 1.
+    6. Get sensor reading.
+    7. Validate that reading from Step 4 doesn't equal reading from Step 6.
+    """
+    print("Step 1", "Set sensor reading interval to 1")
+    new_sensor_reading_interval = 1
+    set_sensor_reading_interval(new_sensor_reading_interval)
+
+    print("Step 2", "Get sensor info")
+    sensor_info_after_reading_interval_change = get_sensor_info()
+
+    print("Step 3", "Validate that sensor reading interval is set to interval from Step 1")
+    assert sensor_info_after_reading_interval_change.get("reading_interval") == new_sensor_reading_interval
+
+    print("Step 4", "Get sensor reading")
+    sensor_reading_before_wait = get_sensor_reading()
+
+    print("Step 5", "Wait for interval specified in Step 1")
+    time.sleep(new_sensor_reading_interval)
+
+    print("Step 6", "Get sensor reading")
+    sensor_reading_after_wait = get_sensor_reading()
+
+    print("Step 7", "Validate that reading from Step 4 doesn't equal reading from Step 6")
+    assert sensor_reading_after_wait != sensor_reading_before_wait
+
+
+# Максимальна версія прошивки сенсора -- 15
+def test_update_sensor_firmware(get_sensor_info, update_sensor_firmware):
+    """
+    1. Get original sensor firmware version.
+    2. Request firmware update.
+    3. Get current sensor firmware version.
+    4. Validate that current firmware version is +1 to original firmware version.
+    5. Repeat steps 1-4 until sensor is at max_firmware_version - 1.
+    6. Update sensor to max firmware version.
+    7. Validate that sensor is at max firmware version.
+    8. Request another firmware update.
+    9. Validate that sensor doesn't update and responds appropriately.
+    10. Validate that sensor firmware version doesn't change if it's at maximum value.
+    """
+    max_firmware_version = 15
+
+    print("Step 1" "Get original sensor firmware version")
+    current_sensor_fw_ver = get_sensor_info().get("firmware_version")
+
+    print("Step 2" "Request firmware update"
+          "Step 3" "Get current sensor firmware version"
+          "Step 4" "Validate that current firmware version is +1 to original firmware version"
+          "Step 5" "Repeat steps 1-4 until sensor is at max_firmware_version - 1")
+    while current_sensor_fw_ver < max_firmware_version - 1:
+        sensor_fw_ver_before_update = current_sensor_fw_ver  # Variable was added to satisfy the requirement in Step 4
+        sensor_fw_upgrade_request = update_sensor_firmware()
```

Comment:
I'd say this is more of a `response_to_update_request`. Also, in the code "update" and "upgrade" seem to be used; it's better to stick to just one word.

Reply:
fixed  - changed to "update"